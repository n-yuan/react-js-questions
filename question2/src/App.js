import React from 'react';

function App() {
  return (
    <main>
      <header>
        <h1>StarRez: Question 2</h1>
      </header>

      <article>
        <p>Render the contents of the data.JSON file</p>
        {/* 
        
        Please see the readme.md file for question information
          
        */}
      </article>
    </main>
  );
}

export default App;
