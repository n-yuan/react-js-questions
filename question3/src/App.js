import React from 'react';

export default () => {

  const handleFormPost = e => {
    alert(JSON.stringify(e))
  }

  return (
    <main>
      <header>
        <h1>StarRez: Question 3</h1>
      </header>

      <article>
        <p>Create an accessible form</p>

        {/* See README.md file for more information */}

      </article>
    </main>
  );
}
