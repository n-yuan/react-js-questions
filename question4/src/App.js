import React, { useState } from 'react';

export default ({ items }) => {
  const [clicked, setClicked] = useState("none")

  return (
    <div>
      <header>
        <h1>StarRez: Question 4</h1>
      </header>

      <article>
        <p>Write some tests for the below functionality</p>

        <h2>{clicked}</h2>

        <ul>
          {items.map(e =>

            <li key={e.name}>
              <button onClick={() => setClicked(e.name)}>{e.name}</button>
            </li>
          )}
        </ul>
      </article>
    </div>
  );
}
