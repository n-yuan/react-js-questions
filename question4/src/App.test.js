import React from 'react';
import App from "./App";
import EnzymeAdapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
import * as enzyme from "enzyme";
import { render } from "@testing-library/react";

enzyme.configure({ adapter: new EnzymeAdapter() });

it('renders without crashing: Enzyme', () => {
  const wrapper = enzyme.mount(
    <App items={[{ name: "one" }]} />,
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});


test("Given CanEdit is false then edit button is not shown", () => {
  const { container, unmount } = render(<App items={[{ name: "one" }]} />);
  expect(container).toMatchSnapshot();
  unmount();
});

// use either enzyme or react testing library