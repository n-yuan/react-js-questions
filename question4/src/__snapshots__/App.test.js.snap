// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given CanEdit is false then edit button is not shown 1`] = `
<div>
  <div>
    <header>
      <h1>
        StarRez: Question 4
      </h1>
    </header>
    <article>
      <p>
        Write some tests for the below functionality
      </p>
      <h2>
        none
      </h2>
      <ul>
        <li>
          <button>
            one
          </button>
        </li>
      </ul>
    </article>
  </div>
</div>
`;

exports[`renders without crashing: Enzyme 1`] = `
<_default
  items={
    Array [
      Object {
        "name": "one",
      },
    ]
  }
>
  <div>
    <header>
      <h1>
        StarRez: Question 4
      </h1>
    </header>
    <article>
      <p>
        Write some tests for the below functionality
      </p>
      <h2>
        none
      </h2>
      <ul>
        <li
          key="one"
        >
          <button
            onClick={[Function]}
          >
            one
          </button>
        </li>
      </ul>
    </article>
  </div>
</_default>
`;
